src/blueprint.zig

```diff
@@ -105,9 +105,48 @@ pub const Blueprint = struct {
 		}
 		return .{.success = self};
 	}
+
+	pub const PasteMode = enum {all, degradable};
+
+	pub fn pasteInGeneration(self: Blueprint, pos: Vec3i, chunk: *ServerChunk, mode: PasteMode) void {
+		switch(mode) {
+			inline else => |comptimeMode| _pasteInGeneration(self, pos, chunk, comptimeMode),
+		}
+	}
+
+	fn _pasteInGeneration(self: Blueprint, pos: Vec3i, chunk: *ServerChunk, comptime mode: PasteMode) void {
+		const chunkOffset = @max(@as(Vec3i, @splat(0)), pos);
+
+		const indexEndX: i32 = @min(@as(i32, chunk.super.width) - chunkOffset[0], @as(i32, @intCast(self.blocks.width)));
```

Comment:
I think it should be `pos` here, instead of `chunkOffset`, otherwise it will be wrong if the structure is bigger than the chunk.

Reply:
Sure.